cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1803,37 +1805,63 @@ class AzureFileSystem::Impl {
                            const AzureLocation& location, bool recursive) {
     DCHECK(!location.container.empty());
     DCHECK(!location.path.empty());
-    // Non-recursive CreateDir calls require the parent directory to exist.
-    if (!recursive) {
+    if (recursive) {
+      std::vector<AzureLocation> target_locations;
+      // Recursive CreateDir calls require there is no file in parents.
```

Comment:
clearer wording:

`// Recursive CreateDir calls require that all path segments be either a directory or not found`

Reply:
Thanks! Applied!